### PR TITLE
[ENH] Add TsGroup.merge_group method

### PR DESCRIPTION
**Pull request by qian-chu**
Implements https://github.com/pynapple-org/pynapple/issues/251.

Added a static method `TsGroup.merge_group()` as well as a `tsgroup.merge()`. Documentation with examples is included as well.

### CODE REVIEW

File: pynapple/core/ts_group.py
Comment (BalzaniEdoardo) [inline pynapple/core/ts_group.py:1123]:
```suggestion
            raise TypeError(f"Input at positions {not_tsgroup_index} are not TsGroup!")
```

File: pynapple/core/ts_group.py
Comment (BalzaniEdoardo) [inline pynapple/core/ts_group.py:1132]: Could you run black on this, it should reformat the spacing

File: pynapple/core/ts_group.py
Comment (BalzaniEdoardo) [inline pynapple/core/ts_group.py:1138]:
```suggestion
                                    "Set `ignore_metadata=True` to bypass the check.")
```

File: pynapple/core/ts_group.py
Comment (BalzaniEdoardo) [inline pynapple/core/ts_group.py:1151]: You could be less "stringent" using `np.allclose` with `atol = 10**(-nap.config.time_index_precision)` and `rtol=0`. @gviejo, what do you think?

File: pynapple/core/ts_group.py
Comment (BalzaniEdoardo) [inline pynapple/core/ts_group.py:1180]: since this is probably the most user facing method, I would have a nice numpydoc docstring here too!

File: pynapple/core/ts_group.py
Comment (BalzaniEdoardo) [inline pynapple/core/ts_group.py:1185]:
```suggestion
        """
        return TsGroup.merge_group(
            self, *tsgroups, reset_index=reset_index, reset_time_support=reset_time_support, ignore_metadata=ignore_metadata)
```

**Review by BalzaniEdoardo — CHANGES_REQUESTED** [review]
Hi Qian-Chu. 
Thanks for your contribution, this is great. Very neat docstrings, and well thought out method. I would ask you a couple of edits before we can merge this:

1. Run linters and formatters
```
black pynapple/core/ts_group.py
isort --check pynapple --profile black
flake8 pynapple --max-complexity 10    
```
2. Add tests of your method hitting all the exceptions. you can use `@pytest.mark.parametrize` to loop over multiple tsgroup and kwargs configs.

Cheers,
Edoardo

File: pynapple/core/ts_group.py
Comment (qian-chu) [reply pynapple/core/ts_group.py:1151]: I thought about `np.allclose` too. But to me the most ideal scenario of setting `reset_time_support=False` would be when the tsgroups were initiated with the same `time_support` and it's hard to imagine how small errors would arise from that. But open to your opinions too!

**Comment by qian-chu** [comment]
> Hi Qian-Chu. Thanks for your contribution, this is great. Very neat docstrings, and well thought out method. I would ask you a couple of edits before we can merge this:
> 
> 1. Run linters and formatters
> 
> ```
> black pynapple/core/ts_group.py
> isort --check pynapple --profile black
> flake8 pynapple --max-complexity 10    
> ```
> 
> 2. Add tests of your method hitting all the exceptions. you can use `@pytest.mark.parametrize` to loop over multiple tsgroup and kwargs configs.
> 
> Cheers, Edoardo

Will do! Thanks for reviewing and providing feedback!

File: pynapple/core/ts_group.py
Comment (gviejo) [reply pynapple/core/ts_group.py:1151]: Yes use allclose.
Thanks

File: pynapple/core/ts_group.py
Comment (qian-chu) [reply pynapple/core/ts_group.py:1180]: That's a good suggestion. Could you offer some tips on how to do so while reducing redundancy?

File: tests/test_ts_group.py
Comment (qian-chu) [inline tests/test_ts_group.py:588]: On my laptop pytest gave file permission error because the npz was not closed during removal. Did a simple fix here.

File: pynapple/core/ts_group.py
Comment (BalzaniEdoardo) [reply pynapple/core/ts_group.py:1191]: I suggest to move the detailed  docstrings from the `merge_group` to the `merge`. 
In `merge_group` keep the same initial paragraph describing the method, the parameters, and the returns sections. I'll add a `See Also` section that points to `merge`.

File: tests/test_ts_group.py
Comment (BalzaniEdoardo) [inline tests/test_ts_group.py:762]:
```suggestion
        with pytest.raises(TypeError,  match=f"Input at positions {[2, 3]} are not TsGroup!"):
            nap.TsGroup.merge_group(ts_group, str, dict)
```

**Review by BalzaniEdoardo — CHANGES_REQUESTED** [review]
Good job!  The only thing left to do is basically imporoving the docstrings of `merge` and then it is good to go!

File: pynapple/core/ts_group.py
Comment (BalzaniEdoardo) [inline pynapple/core/ts_group.py:1112]:
```suggestion
            Index    rate
            -------  ------
                  0     0.3
                 10     0.3
```

Otherwise, it will be rendered as a table in the docs.

File: pynapple/core/ts_group.py
Comment (BalzaniEdoardo) [inline pynapple/core/ts_group.py:1101]:
```suggestion
            Index    rate
            -------  ------
                  0     1.5
                  1     1.5
```
Rendered wrongly by mkdocs otherwise

**Review by BalzaniEdoardo — COMMENTED** [review]
A couple of fixes on how the docstrings are rendered by mkdocs

**Comment by BalzaniEdoardo** [comment]
I re-based this into the `dev` branch, but up to @gviejo.

@qian-chu this is definitely going to be in `main` but we usually have merge first into `dev`, so that we have some time to try out the new feature.

File: tests/test_ts_group.py
Comment (qian-chu) [reply tests/test_ts_group.py]: Interestingly, this change gave a perplexing error

```
E           AssertionError: Regex pattern did not match.
E            Regex: 'Input at positions [2, 3] are not TsGroup!'
E            Input: 'Input at positions [2, 3] are not TsGroup!'
E            Did you mean to `re.escape()` the regex?
```

**Comment by qian-chu** [comment]
Many thanks @BalzaniEdoardo for all the suggestions. I just moved the examples from `merge_group()` to `merge()`. Feel free to take a look. Not sure how to implement See Also `merge_group()` as a link but I did leave a mentioning there :)

I have no problem with `dev` vs `main`! Take your time.

File: tests/test_ts_group.py
Comment (qian-chu) [reply tests/test_ts_group.py]: Fixed with replacing `{[2, 3]}` with `(.*)`

**Comment by BalzaniEdoardo** [comment]
> Many thanks @BalzaniEdoardo for all the suggestions. I just moved the examples from `merge_group()` to `merge()`. Feel free to take a look. Not sure how to implement See Also `merge_group()` as a link but I did leave a mentioning there :)
> 
> I have no problem with `dev` vs `main`! Take your time.

Great, I'll add the link, no worries!

**Review by BalzaniEdoardo — APPROVED** [review]

**Review by gviejo — APPROVED** [review]

**Comment by gviejo** [comment]
Great PR. Thank you @qian-chu

**Pull request merged**
